src/Mvc/Mvc.NewtonsoftJson/src/BsonTempDataSerializer.cs

```diff
@@ -208,6 +210,8 @@ private static bool CanSerializeType(Type typeToSerialize, out string errorMessa
             }
 
             actualType = actualType ?? typeToSerialize;
+            actualType = Nullable.GetUnderlyingType(actualType) ?? actualType;
```

Comment:
Is this a new issue found while testing?

Reply:
Yup and it's mostly my fault. We were previously unwrapping as part of the filter - https://github.com/aspnet/AspNetCore/blame/d116ab8550ea67aefd67992072cb3bfe08723f0c/src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Filters/SaveTempDataPropertyFilterBase.cs#L125, but this got missed when I changed it from a static method on the one `TempDataSerializer` type to an instance on `TempDataSerializer`. Added some tests that cover this